catlin/pkg/validator/task_validator.go

```diff
@@ -112,6 +113,17 @@ func (t *taskValidator) validateStep(s v1beta1.Step) Result {
 		result.Warn("Step %q uses secret as environment variables. Prefer using secrets as files over secrets as environment variables", step)
 	}
 
+	if s.Script != "" {
+		expr, _ := extractExpressionFromString(s.Script, "params")
+		if expr != "" {
+			result.Warn(
+				"Step %q references %q directly from its script block. Consider putting the param into an environment variable of the Step and accessing that environment variable in your script instead.",
```

Comment:
I wonder if it's worthwhile including the reasoning for this warning, such as:

```suggestion
				"Step %q references %q directly from its script block. For reliability and security, consider putting the param into an environment variable of the Step and accessing that environment variable in your script instead.",
```